flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/operations/SqlToOperationConverter.java

```diff
@@ -279,8 +278,61 @@ private Operation convertAlterTable(SqlAlterTable sqlAlterTable) {
 				throw new ValidationException(String.format("Table %s doesn't exist or is a temporary table.",
 					tableIdentifier.toString()));
 			}
+		} else if (sqlAlterTable instanceof SqlAlterTableAddConstraint) {
+			Optional<CatalogManager.TableLookupResult> optionalCatalogTable =
+					catalogManager.getTable(tableIdentifier);
+			if (optionalCatalogTable.isPresent() && !optionalCatalogTable.get().isTemporary()) {
+				SqlTableConstraint constraint = ((SqlAlterTableAddConstraint) sqlAlterTable)
+						.getConstraint();
+				validateTableConstraint(constraint);
+				CatalogTable oriCatalogTable = (CatalogTable) optionalCatalogTable.get().getTable();
+				TableSchema.Builder builder = TableSchemaUtils
+						.builderWithGivenSchema(oriCatalogTable.getSchema());
+				if (constraint.getConstraintName() != null) {
+					builder.primaryKey(constraint.getConstraintName(), constraint.getColumnNames());
```

Comment:
If we are going to set not null in `builder.primaryKey`, then we should check the key fields not null before calling the method. Because FLIP-84 says:

> In alter statements if a users says to create a primary key on a nullable column the operation should fail.